### PR TITLE
add change password field for users/admins

### PR DESCRIPTION
**Pull request by jsaalfeld**

### CODE REVIEW

File: client/src/app/site/users/components/user-detail/user-detail.component.ts [hunk 1]
```diff
@@ -322,6 +331,25 @@ export class UserDetailComponent extends BaseViewComponent implements OnInit {
...
+
+    /**
+     * Sends new Password entered in the new password field to server
+     */
+    public saveNewPassword(): Promise<void> {
+        this.personalInfoForm.get('new_password').disable();
+        const password = this.personalInfoForm.get('new_password').value;
+        this.personalInfoForm.get('new_password').reset();
+        this.newPassword = false;
+        return this.repo.resetPassword(this.user, password);
```
Comment (FinnStutzenstein) [inline client/src/app/site/users/components/user-detail/user-detail.component.ts]: remove the return, if not needed by the calling function

**Comment by FinnStutzenstein** [comment]
Yes, this should go into an own view. Also there are not permission checks, because everyone logged in should be able to change his password.

**Comment by jsaalfeld** [comment]
Can we discuss this tomorrow in person?

**Comment by jsaalfeld** [comment]
added the requested changes @tsiegleauq & @FinnStutzenstein

**Comment by jsaalfeld** [comment]
I don't get the error in the UserDetailComponent...

The Debugging-Output states
`Uncaught Error: Uncaught (in promise): Error: Cannot find module './site/site.module`

There is nothing i can change or do to prevent it from failing. Maybe @FinnStutzenstein  or @tsiegleauq ?

File: client/src/app/site/users/services/user-repository.service.ts [hunk 2]
```diff
@@ -110,17 +111,28 @@ export class UserRepositoryService extends BaseRepository<ViewUser, User> {
...
      */
     public async resetPassword(user: ViewUser, password: string): Promise<void> {
-        await this.update({ default_password: password }, user);
         const path = `/rest/users/user/${user.id}/reset_password/`;
         await this.httpService.post(path, { password: password });
     }
 
+    /**
+     * Updates the password and sets a new one, if the old one was correct.
+     * @param oldPassword the old password
```
Comment (tsiegleauq) [inline client/src/app/site/users/services/user-repository.service.ts]: missing empty line

File: client/src/app/site/users/components/password/password.component.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    public onKeyDown(event: KeyboardEvent): void {
+        if (event.key === 'Enter' && event.shiftKey) {
+            this.save();
+        }
+    }
+
+    /**
+     * Initializes the forms and some of the frontend options
+     */
+    public ngOnInit(): void {
```
Comment (tsiegleauq) [inline client/src/app/site/users/components/password/password.component.ts]: belongs under constructor

File: client/src/app/site/users/components/password/password.component.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+import { TranslateService } from '@ngx-translate/core';
+import { MatSnackBar } from '@angular/material';
+import { Title } from '@angular/platform-browser';
+
+import { ViewUser } from '../../models/view-user';
+import { UserRepositoryService } from '../../services/user-repository.service';
+import { OperatorService } from '../../../../core/services/operator.service';
+import { BaseViewComponent } from '../../../../site/base/base-view';
+
+@Component({
```
Comment (tsiegleauq) [inline client/src/app/site/users/components/password/password.component.ts]: missing doc

**Comment by tsiegleauq** [comment]
plus: the close-button on change password field does nothing right now.

**Comment by jsaalfeld** [comment]
resolved everything

**Comment by tsiegleauq** [comment]
took care of the tests.

**Pull request merged**